docs/Guide.Jest.md

```diff
@@ -19,32 +19,39 @@ You should remove `e2e/mocha.opts`, you no longer need it.
 ### 3. Write a detox setup file
 
 ```js
+// ./jest/setup/
 const detox = require('detox');
 const config = require('../package.json').detox;
 
 // Set the default timeout
 jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
 
-beforeAll(async () => {
-  await detox.init(config);
-});
+// setup detox only when running e2e tests
+if (process.argv[2].includes('__e2e__')) {
+  beforeAll(async () => {
+    await detox.init(config);
+  });
 
-afterAll(async () => {
-  await detox.cleanup();
-});
+  afterAll(async () => {
+    await detox.cleanup();
+  });
 
-// optional, you may remove this part
-beforeEach(async () => {
-  await device.reloadReactNative();
-});
+  beforeEach(async () => {
+    await device.reloadReactNative();
+  });
+}
 ```
 
 ### 4. Run jest
 
 Add this part to your `package.json`:
 ```json
+"jest": {
+  "preset": "react-native",
+  "setupTestFrameworkScriptFile": "<rootDir>/jest/setup.js"
+},
 "scripts": {
-    "test:e2e": "jest e2e --setupTestFrameworkScriptFile=./jest/setup-e2e-tests.js --runInBand"
+    "test:e2e": "detox build && jest __e2e__ --runInBand"
```

Comment:
The build part is left out because with `detox test` it isn't included either, but adding something like `test:e2e:build: "detox build"` is a pretty neat idea!